clients/spark/core/src/test/scala/io/treeverse/jpebble/BlockParserSpec.scala

```diff
@@ -287,71 +300,101 @@ class GolangContainerSpec extends AnyFunSpec with ForAllTestContainer {
-        "fuzz.contents.0",
+        "fuzz.contents.0.with.shared.prefix",
         "fuzz.contents.1",
-        "fuzz.contents.2",
+        "fuzz.contents.2.with.shared.prefix",
         "fuzz.contents.3",
-        "fuzz.contents.4",
+        "fuzz.contents.4.with.shared.prefix",
         "fuzz.contents.5"
       )
 
       testFiles.foreach(fileName =>
         describe(fileName) {
           it("should parse successfully") {
-            withGeneratedTestFiles(fileName, verifyBlockParserOutput)
+            withGeneratedSstTestFiles(fileName, verifyBlockParserOutput)
           }
         })
     }
 
     describe("with random table user properties") {
       it("should parse successfully") {
-        withGeneratedTestFiles("fuzz.table.properties", verifyBlockParserOutput)
+        withGeneratedSstTestFiles("fuzz.table.properties", verifyBlockParserOutput)
       }
     }
 
     describe("with sstable with xxHash64 checksum") {
       it("should fail parsing") {
         assertThrows[BadFileFormatException]{
-          withGeneratedTestFiles("checksum.type.xxHash64", verifyBlockParserOutput)
+          withGeneratedSstTestFiles("checksum.type.xxHash64", verifyBlockParserOutput)
         }
       }
     }
 
     describe("with sstable with levelDB table format") {
       it("should fail parsing") {
         assertThrows[BadFileFormatException]{
-          withGeneratedTestFiles("table.format.leveldb", verifyBlockParserOutput)
+          withGeneratedSstTestFiles("table.format.leveldb", verifyBlockParserOutput)
+        }
+      }
+    }
+
+    describe("with sstable with Zstd compression") {
+      it("should fail parsing") {
+        assertThrows[BadFileFormatException]{
+          withGeneratedSstTestFiles("compression.type.zstd", verifyBlockParserOutput)
         }
       }
     }
 
     describe("with max size sstable supported by lakeFS") {
       it("should parse successfully") {
-        withGeneratedTestFiles("max.size.lakefs.file", verifyBlockParserOutput)
+        withGeneratedSstTestFiles("max.size.lakefs.file", verifyBlockParserOutput)
       }
     }
 
     describe("with random restart interval") {
       it("should parse successfully") {
-        withGeneratedTestFiles("fuzz.block.restart.interval", verifyBlockParserOutput)
+        withGeneratedSstTestFiles("fuzz.block.restart.interval", verifyBlockParserOutput)
       }
     }
 
     describe("with random block size") {
       it("should parse successfully") {
-        withGeneratedTestFiles("fuzz.block.size", verifyBlockParserOutput)
+        withGeneratedSstTestFiles("fuzz.block.size", verifyBlockParserOutput)
       }
     }
 
     describe("with random blocksize threshold") {
       it("should parse successfully") {
-        withGeneratedTestFiles("fuzz.block.size.threshold", verifyBlockParserOutput)
+        withGeneratedSstTestFiles("fuzz.block.size.threshold", verifyBlockParserOutput)
+      }
+    }
+
+    describe("with non-sst file of size 0") {
```

Comment:
```suggestion
    describe("with empty file") {
```
(also rename the generator ?)